src/main/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeActor.scala

```diff
@@ -1,79 +1,260 @@
-    new ExportEntitiesByTypeActor(app.rawlsDAO, userInfo)
+  def constructor(app: Application, materializer: ActorMaterializer)(userInfo: UserInfo)(implicit executionContext: ExecutionContext) =
+    new ExportEntitiesByTypeActor(app.rawlsDAO, userInfo, materializer)
 }
 
-class ExportEntitiesByTypeActor(val rawlsDAO: RawlsDAO, val userInfo: UserInfo)(implicit protected val executionContext: ExecutionContext) extends Actor with ExportEntitiesByType {
+class ExportEntitiesByTypeActor(val rawlsDAO: RawlsDAO, val argUserInfo: UserInfo, argMaterializer: ActorMaterializer)(implicit protected val executionContext: ExecutionContext) extends Actor with LazyLogging {
+
+  // Requires its own actor context to work with downstream actors: TSVWriterActor and StreamingActor
+  def actorRefFactory: ActorContext = context
+
+  implicit val timeout: Timeout = Timeout(1 minute)
+  implicit val userInfo: UserInfo = argUserInfo
+  implicit val materializer: ActorMaterializer = argMaterializer
+
   override def receive: Receive = {
-    case ExportEntities(workspaceNamespace, workspaceName, filename, entityType, attributeNames) => exportEntities(workspaceNamespace, workspaceName, filename, entityType, attributeNames) pipeTo sender
+    case ExportEntities(ctx, workspaceNamespace, workspaceName, entityType, attributeNames) => streamEntities(ctx, workspaceNamespace, workspaceName, entityType, attributeNames) pipeTo sender
+  }
+
+  /**
+    * Two basic code paths
+    *
+    * For Collection types, write the content to temp files, zip and return.
+    *
+    * For Singular types, pipe the content from `Source` -> `Flow` -> `Sink`
+    * Source generates the entity queries
+    * Flow executes the queries and sends formatted content to chunked response handler
+    * Sink finishes the execution pipeline
+    *
+    * Handle exceptions directly by completing the request.
+    */
+  def streamEntities(ctx: RequestContext, workspaceNamespace: String, workspaceName: String, entityType: String, attributeNames: Option[IndexedSeq[String]]): Future[Unit] = {
+    getEntityTypeMetadata(workspaceNamespace, workspaceName, entityType) flatMap { metadata =>
+      val entityQueries = getEntityQueries(metadata, entityType)
+
+      // Stream exceptions have to be handled by directly closing out the RequestContext responder
+      if (TSVFormatter.isCollectionType(entityType)) {
+        streamCollectionType(ctx, workspaceNamespace, workspaceName, entityType, entityQueries, metadata, attributeNames).onFailure {
+          case t: Throwable => handleStreamException(ctx, t)
+        }
+      } else {
+        val headers = TSVFormatter.makeEntityHeaders(entityType, metadata.attributeNames, attributeNames)
+        streamSingularType(ctx, workspaceNamespace, workspaceName, entityType, entityQueries, metadata, headers, attributeNames).onFailure {
+          case t: Throwable => handleStreamException(ctx, t)
+        }
+      }
+      Future(())
```

Comment:
I'm not sure this does what you want -- what's happening here is you're creating 2 _independent_ futures and only returning the second one. Thus the Future returned by this method is not attached in any way to the Futures returned by `streamCollectionType`/`streamSingularType`.

Maybe something like this would be preferable?
```
def streamEntities(ctx: RequestContext, workspaceNamespace: String, workspaceName: String, entityType: String, attributeNames: Option[IndexedSeq[String]]): Future[Unit] = {
    getEntityTypeMetadata(workspaceNamespace, workspaceName, entityType) flatMap { metadata =>
      val entityQueries = getEntityQueries(metadata, entityType)

      // Stream exceptions have to be handled by directly closing out the RequestContext responder
      val streamFuture = if (TSVFormatter.isCollectionType(entityType)) {
        streamCollectionType(ctx, workspaceNamespace, workspaceName, entityType, entityQueries, metadata, attributeNames)
      } else {
        val headers = TSVFormatter.makeEntityHeaders(entityType, metadata.attributeNames, attributeNames)
        streamSingularType(ctx, workspaceNamespace, workspaceName, entityType, entityQueries, metadata, headers, attributeNames)
      }

      streamFuture.mapTo[Unit].recover { case t: Throwable =>
        handleStreamException(ctx, t)
      }
    }
  }.recoverWith {
    // Standard exceptions have to be handled as a completed request
    case t: Throwable => handleStandardException(ctx, t)
  }
```